File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    hash_dict = {}

    for p in participant:
        hash_dict[p] = hash_dict.get(p, 0) + 1
        
    for c in completion:
        hash_dict[c] -= 1
        
    for key in hash_dict:
        if hash_dict[key] > 0:
            return key

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

// 선수 100,000 = 시간 제한
string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> player;
    
    // 테이블 생성
    for (string p : participant) player[p] += 1;
    
    // 동명이인 같은 키 값
    for (string p : completion) {
        player[p] -= 1;
        if (player[p] == 0) {
            player.erase(player.find(p));
        }        
    }
    return player.begin()->first;
}